pkg/hostpath/hostpath.go

```diff
@@ -166,28 +167,38 @@ func (hp *hostPath) getVolumePath(volID string, volCtx map[string]string) string
 	return filepath.Join(hp.root, volID, podNamespace, podName, podUID, podSA)
 }
 
-func createFile(path string, buf []byte) {
-	file, err := os.Create(path)
-	if err != nil {
-		klog.Errorf("error creating file %s: %s", path, err.Error())
-		return
-	}
-	defer file.Close()
-	file.Write(buf)
-}
-
 func commonUpsertRanger(podPath, filter string, key, value interface{}) bool {
 	if key != filter {
 		return true
 	}
-	buf, err := json.MarshalIndent(value, "", "    ")
-	if err != nil {
-		klog.Errorf("error marshalling: %s", err.Error())
+	payload, _ := value.(Payload)
+	podFileDir := filepath.Join(podPath, fmt.Sprintf("%s", key))
+	if err := os.MkdirAll(podFileDir, os.ModePerm); err != nil {
+		klog.Errorf("unable to make dir %s: %s", podFileDir, err.Error())
 		return true
```

Comment:
Should this return false since it's a failure?

Reply:
Yeah this stems from how the golang ranger contract works.... there is not error propagation.

To date I've been of the opinion that best effort is sufficient, and that if we are getting golang filesystem errors like this, most likely retry is not going to help, as most likely we are hitting "the host is messed up" errors vs. intermittent RPC over the network type things.  So kind of "special casey".

And I like utilizing golang based API here.

We'll ultimately get a retry when the controller relists the configmaps/secrets anyway.... though at a much more gradual interval.

That said, I'll think about it some more, and see if I'm convinced we should pivot at all.  Minimally, I should add comments to capture all this nuance.